backend/common/src/main/java/ai/verta/modeldb/common/db/migration/MigrationTools.java

```diff
@@ -0,0 +1,35 @@
+package ai.verta.modeldb.common.db.migration;
+
+import java.nio.charset.StandardCharsets;
+import java.sql.SQLException;
+import java.util.zip.CRC32;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class MigrationTools {
```

Comment:
But, this should be an internal implementation detail, and I should make this utility class non-public, for sure.

Reply:
I'll make sure to do that in a follow-on PR, since I've got two branches I'd need to update with this change, otherwise.

Reply:
Just as a note in favor of not having a class hierarchy... the H2 implementation doesn't need to actually do any real database locking...just a local lock, since the db is in-memory. That would mean that the H2 implementation had a method attached to it that wasn't relevant to its job at all.

Reply:
I agree that class hierarchies can get out of hand really fast.  But avoiding them completely (while still using interfaces) results situations like this where you have a utility class whose methods are really only used in one scenario.  The `MigrationDatastore` interface might as well be the internal too since we provide implementations for all the database types that we support.  But older Java only supports public interfaces...

Reply:
Can definitely make the interface non-public and am happy to do so.